axum-extra/src/middleware.rs

```diff
@@ -0,0 +1,44 @@
+//! Additional middleware utilities.
+
+use crate::either::Either;
+use tower_layer::Identity;
+
+/// Convert an `Option<Layer>` into a [`Layer`].
+///
+/// If the layer is a `Some` it'll be applied, otherwise not.
+///
+/// # Example
+///
+/// ```
+/// use axum_extra::middleware::option_layer;
+/// use axum::{Router, routing::get};
+/// use std::time::Duration;
+/// use tower_http::timeout::TimeoutLayer;
+///
+/// # let option_timeout = Some(Duration::new(10, 0));
+/// let timeout_layer = option_timeout.map(TimeoutLayer::new);
+///
+/// let app = Router::new()
+///     .route("/", get(|| async {}))
+///     .layer(option_layer(timeout_layer));
```

Comment:
Just to be sure, if we pass `.layer(Identity::new())`, it means zero additional cost for each incoming request?
Just like we wouldn't write `.layer(...)` at all?

Reply:
It does have additional cost. It'll create a new [`Route`](https://github.com/tokio-rs/axum/blob/main/axum/src/routing/route.rs#L30) which requires allocation and a dynamic dispatch. I doubt that impact will be measurable in a real world app however but if you really wanna avoid it you have to use an `if`.

Reply:
And by the way this is why we recommend adding multiple middleware with [`ServiceBuilder`](https://docs.rs/axum/latest/axum/middleware/index.html#applying-multiple-middleware).